spark/src/main/resources/error/delta-error-classes.json

```diff
@@ -529,6 +529,21 @@
     ],
     "sqlState" : "42710"
   },
+  "DELTA_CONSTRAINT_DATA_TYPE_MISMATCH" : {
+    "message" : [
+      "Column <columnName> has data type <columnType> and cannot accept data type <dataType> because this column is referenced by the following check constraint(s):",
```

Comment:
I assume the contraints can already be retrieved by the user and we are not introducing a way to read them now?

Reply:
You can display constraints e.g. using DESCRIBE DETAIL when you have SELECT permissions on a table so users already have access to that information (also another error already displays the constraints)